### ISSUE TITLE
Add support to goto hunk locations from commit view

### ISSUE THREAD

**Issue by printfdebugging** [NeogitOrg/neogit#1239]
**Is your feature request related to a problem? Please describe.**
I contribute to libreoffice quite often, and therefore have to commit my changes and push them as patches.
Now when I want to go back to some hunk in the commit, and modify it, I cannot do it from the commit  view.

**Describe the solution you'd like**
I would like the commit view to behave like the views for unstaged and untracked changes, where pressing enter takes
me to the change

**Comment by printfdebugging** [comment]
and I would like to work on this one :)

**Comment by printfdebugging** [comment]
We don't have a bot here I guess.

**Comment by CKolkey** [comment]
Beep boop... er, no, no bot, haha. But happy to help however I can :)

I -highly- suggest you base on the `nightly` branch, if you haven't already done so. It still works with nvim 0.9, but has a bunch of new stuff for 0.10

**Comment by printfdebugging** [comment]
@CKolkey Yes, thanks for reminding, I was using neovim from package manager :). This is the right time to have a local copy of my own.

**Comment by printfdebugging** [comment]
I will take some time(3-4 days min). This is a really cool plugin, and I don't want to just copy and paste all the functions from `status/ `to `commit_view/`,  and call it a day. Plan is to read it through, and learn how it does what it does in depth(would be more than a crash course to neovim lua plugin developement). Then I will refactor the common functions, and send it for review.

**Comment by CKolkey** [comment]
No rush - one thing I'm wondering: when you `goto` the hunk/file from the commit view, are you imaging it being from `HEAD` or at the commit you're looking at?

**Comment by printfdebugging** [comment]
Does it matter?
I don't know how we look for the specific hunk location in the files yet, but if it is purely using line number from the diffs,
then it should be something like "Take me to this line number of the file, if the file exists, and if this line number is in the bounds".
**yth line in file x on `HEAD`**

**Comment by CKolkey** [comment]
Might be cool to have both, actually. Something like `<cr>` going to the location on current HEAD (as you describe) and `<s-cr>` checking out the commit first, then opening the file. No worries though, just mess around with it and see what you come up with :)

**Comment by printfdebugging** [comment]
Hi, I realize that I since I don't have any experience with lua, (Just that I read the PIL book), I am not reaching anywhere. Even if I hunt and gather some lines of code from here and there, that will not help much. I do understand the code though like what's going on in the util.lua, but that's not enough to see the whole picture. (eye opener for me)

I will be back to work on this issue, and the other issues, and if in case someone else wants this very bad, they can take it from here (I don't want to leave it, but sometimes a stepback helps take a big leap). Will write some programs of my own, do some playing around. I hope @CKolkey finds it reasonible. Or you can suggest me some approach.

**Comment by CKolkey** [comment]
No worries :) This isn't the simplest codebase to jump into - looking forward to see what you come up with

**Comment by txemaotero** [comment]
Hi guys! 
First, I just want to say: THANK YOU for this plugin! I use it everyday and it feels so good to have something comparable to magit inside neovim (I can finally say to my emacs friends "I've no reason to move to emacs now" 😆 ). 

That said, I was missing this feature a lot in my daily workflow. I very often rely on the CommitView to self-review changes I made and I usually find bugs or things that need to be changed so, being able to jump to the code directly from that view is a big win. As there wasn't much activity in this issue, I decided to give it a try and implement it myself (see the open PR). I've been using the feature at my job these days while still developing and I really enjoyed the experience, it's exactly what I needed.

This is my first contribution to the project so most certainly there are mistakes or missing things, just let me know and I'll address any suggestion with pleasure 😃

**Comment by XeroOl** [comment]
It seems like it's not possible to set `<cr>` to be the sane behavior.
`opts = {mappings = {commit_view = {["<cr>"] = "OpenFileInWorkTree"}}}` gets overridden and doesn't change the behavior for me.

**Comment by txemaotero** [comment]
Hi @XeroOl , that's correct. You cannot set to `<cr>` because that overlaps with an existing mapping. I've decided to overload  `<cr>` to open the file in the commit because that is the default behavior of magit. However, I can perfectly understand that some (or most, I don't know 🤷‍♂️ ) people may prefer to have `<cr>` to jump to the file in the current version and the other mapping (`a` by default) to open the file in the corresponding commit. That's why I've decided to introduce a configuration option to interchange the behavior of `<cr>` and `a`. This is the [PR](https://github.com/NeogitOrg/neogit/pull/1866) with the changes. Once merged, you should be able to set: `opts = {commit_view = {invert_open_file = false}` and have the sane behavior 😉
